### PR TITLE
Port of /tg/station's Shadowlings [WIP]

### PR DESCRIPTION
**Pull request by TheDZD**
# SOON™
Ports tgstation/-tg-station#8157

This won't compile, needs to be fixed and tested. The only reason I am PRing  this right now is so that we don't have multiple people working on the same thing.

@ZomgPonies This will require PROBABILITY SHADOWLING # to be added to game_options.txt, and will likely require SQL database additions for the BE_SHADOWLING antag preference.

## To-do
- [x] Make it compile.
- [x] Port tgstation/-tg-station#8927.
- [x] Test for runtimes.
- [x] Make the Shadowling hivemind a language instead of a verb.
- [x] Fix [these](http://hastebin.com/raw/pucoconese) runtimes, ~~somehow~~ with hacks!
- [ ] Test for bugs.
- [ ] Test to see if the game mode works.
- [ ] Fix bugs/game mode issues.
- [ ] Make wiki paged based on [/tg/'s](https://tgstation13.org/wiki/Shadowling)

### CODE REVIEW

**Comment by Fishocalypse** [comment]
TRAVIS PLS :+1: :100: 

MERGE ASAP MERGE NOW AAAAA

**Comment by Spacemanspark** [comment]
YES YES YES.

**Comment by Finalsong** [comment]
MERGE DIS FUKKEN SHIT

**Comment by ZomgPonies** [comment]
@TheDZD It actually doesn't require any database manipulation at all. You'll probably need to add BE_SHADOWLING to the list in setup.dm, and of course fix up preferences.dm if not already done.

**Comment by ZomgPonies** [comment]
Also port https://github.com/tgstation/-tg-station/pull/8927

**Comment by Incoming5643** [comment]
You guys sure you want to port the mode while it's still being actively tweaked? You'll probably need to port more fixes later.

**Comment by Fox-McCloud** [comment]
@Incoming5643 

We can keep up; if it turns out to be more problematic and it's best for us to just "wait", then we can easily disable it in our config so it doesn't show up unless we want it to.

File: code/game/gamemodes/shadowling/shadowling.dm [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+
+/datum/game_mode/shadowling
+	name = "shadowling"
+	config_tag = "shadowling"
+	required_players = 30
+	required_enemies = 2
+	recommended_enemies = 2
+	restricted_jobs = list("AI", "Cyborg")
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain")
```
Comment (Fishocalypse) [inline code/game/gamemodes/shadowling/shadowling.dm]: Suggesting adding our snowflake jobs(such as Blueshield, NT rep, IAA, and the Sec pod pilot in.) to the job list.

**Comment by ZomgPonies** [comment]
Okay from what I under this uses the 'shadowpeople' mutation? Let me get to work updating it to BS12 standard right now (meaning as a full human sub-species), and then we can work integrating this PR into it properly. If we're going to add more species (even 'event' species like shadowlings or golems) we should do it right.

**Comment by ZomgPonies** [comment]
https://github.com/ParadiseSS13/Paradise/pull/800

- This updates shadowpeople to be a human species. This PR will need to be updated to take that into account. When changing a normal human species to shadowpeople, use the set_species proc.

**Comment by Regen1** [comment]
:+1:  :100: 
I love it

**Comment by tigercat2000** [comment]
@ZomgPonies I am pretty sure [shadowlings have their own species datum](https://github.com/ParadiseSS13/Paradise/pull/796/files#diff-c3c200bb2b2141498fa4896af72a6591R228)

**Comment by TheDZD** [comment]
The Shadowling species datum is supposed to be a child of the shadow person one, although they do need to override their light/darkness interactions, as Shadowlings take significantly more damage from light, but heal faster in the dark and also heal from toxins in the dark.

**Comment by SamCroswell** [comment]
:+1: :heavy_multiplication_x: :100:

**Comment by TheDZD** [comment]
And now, #857 is preventing the shadowling sprite from displaying, yay.

**Comment by TheDZD** [comment]
[These](http://hastebin.com/raw/pucoconese) runtimes make no sense, and they keep happening. If somebody could test to see if they happen when they try spawning brains/posibrains, and turning somebody into a shadowling via player panel, that would be helpful.

And to BYOND, :fu:

**Comment by TheDZD** [comment]
The runtimes are gone. [YES YES!](https://www.youtube.com/watch?v=P3ALwKeSEYs).

**Comment by Ragonath** [comment]
This is going to be amazing when complete. Will you just be porting the gamemode or will you be adding to and tweaking that gamemode untill it has that 'Paradise' touch?

**Comment by TheDZD** [comment]
Probably going to be tweaking it a bit since Xhuis isn't touching the gamemode for a bit, and it is relatively new.

**Comment by TheDZD** [comment]
This might be ready for testing now, for the most part it's a 1:1 port from /tg/, although there are a few differences with the shadowling species.

**Pull request merged**